thirdparty/pybind11/include/pybind11/attr.h

```diff
@@ -12,7 +12,7 @@
 
 #include "cast.h"
 
-NAMESPACE_BEGIN(PYBIND11_NAMESPACE)
+PYBIND11_NAMESPACE_BEGIN(PYBIND11_NAMESPACE)
```

Comment:
could we make this dependency into an external project? there are a few dependencies that are directly committed into `thirdparty`, and I feel like we should minimize their number, this upgrade would be a great excuse to do so with pybind11

Reply:
What do you think about FetchContent? I prefer it over ExternalProject, because ExternalProject downloads sources during build while FetchContent does it during configure, which means if you're using an IDE, headers may not be present and include paths may be broken before the first build. It also results in simpler cmake code IMO.

Done in a2ac1f3, but I couldn't reuse pybind11's own cmake files. They look reasonable, but I just kept getting linker errors that I was unable to solve.

Reply:
looks clean, thank you